pkg/aws/eks/pricing_map.go

```diff
@@ -144,7 +155,12 @@ func weightedPriceForInstance(price float64, attributes Attributes) (*ComputePri
 	if err != nil {
 		return nil, fmt.Errorf("%w: %w", ErrParseAttributes, err)
 	}
-	ratio := cpuToCostRation[attributes.InstanceFamily]
+	ratio := cpuToCostRatio[attributes.InstanceFamily]
+	if ratio == 0 {
+		log.Printf("no ratio found for instance type %s, defaulting to %s", attributes.InstanceType, defaultInstanceFamily)
+		ratio = cpuToCostRatio[defaultInstanceFamily]
+	}
+
```

Comment:
@inkel is there a more idiomatic way of achieving this in Go? Plainly, what I want to do is if the instance family doesn't exist, I want to use a default key instead.

Reply:
The more idiomatic way would be using the following:
```suggestion
	ratio, ok := cpuToCostRatio[attributes.InstanceFamily]
	if !ok {
		log.Printf("no ratio found for instance type %s, defaulting to %s", attributes.InstanceType, defaultInstanceFamily)
		ratio = cpuToCostRatio[defaultInstanceFamily]
	}

```

The reason why is because `0` is the zero value for `float64`, and you could have a element in the map whose value is `0` (whether that's a valid value or not depends on your use case), thus by using the _two-form access_ to the map, `ok` will return if the key was present or not in the map, ej:

```go
m := map[string]int{"zero": 0, "one": 1}
a, ok := m["zero"] // a == 0, ok == true
a, ok = m["cero"] // a == 0, ok == false
```

Basically is a way of checking that the returned value was found in the map and not the zero value of the element type.